src/lib.rs

```diff
@@ -82,6 +89,54 @@ impl Deref for AllocLock {
     }
 }
 
+unsafe impl GlobalAlloc for GCMalloc {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let ptr = libc::malloc(layout.size() as libc::size_t) as *mut u8;
+
+        ALLOC_LOCK.lock();
+        match ALLOC_INFO {
+            Some(ref mut pm) => pm.insert(ptr as usize, layout.size()),
+            None => {
```

Comment:
So this is lazily initialised. I guess you couldn't figure out a way to have it created at the start?

Reply:
Unfortunately not without it becoming part of `rustc`. Ideally this initialisation would happen before `main`.

Reply:
ok

Reply:
`lazy_static` and the like do similar tricks, they just make them a bit less visible.

Reply:
I think there was a reason he couldn't use `lazy_static` too. What was it again, Jake?

Reply:
I decided against `lazy_static` as it obscured the locking logic.

Reply:
Sorry, just to be clear, I agree with not using `lazy_static` -- I was just pointing out the (rough) similarity.